src/exo/LoopIR_scheduling.py

```diff
@@ -194,8 +194,52 @@ def map_e(self,e):
 
         return super().map_e(e)
 
+def get_reads(e):
+    if isinstance(e,LoopIR.Read):
+        return sum([get_reads(e) for e in e.idx], [(e.name, e.type)])
+    elif isinstance(e,LoopIR.USub):
+        return get_reads(e.arg)
+    elif isinstance(e,LoopIR.BinOp):
+        return get_reads(e.lhs) + get_reads(e.rhs)
+    elif isinstance(e,LoopIR.BuiltIn):
+        return sum([get_reads(a) for a in e.args], [])
+    elif isinstance(e,LoopIR.Const):
+        return []
+    else:
+        assert False, "bad case"
+
+class _DoMergeWrites(LoopIR_Rewrite):
+    def __init__(self, proc, stmt1, stmt2):
+        try:
+            assert len(stmt1.idx) == len(stmt2.idx)
+            for i, j in zip(stmt1.idx, stmt2.idx):
+                Check_ExprEqvInContext(proc, [stmt1, stmt2], i, j)
+        except SchedulingError:
+            raise SchedulingError("expected the left hand side's indices to be the same.")
```

Comment:
You are handling an error by raising an error. The right to do that is `raise ... from` to tell the Python runtime that there aren't _two_ errors.
```suggestion
        except SchedulingError as e:
            raise SchedulingError("expected the left hand side's indices to be the same.") from e
```